### DISCUSSION TITLE
Dynamically change alias at runtime

### DISCUSSION THREAD

**Discussion by jesumyip** [pydantic/pydantic#7725] [Question] [answered]
Is this even possible?

```
class Person(BaseModel):
  name: str = Field(alias="common_name")
  age: int = Field(alias="how_old")

data = { "name" : "John", "age" : 200 }
print(data.model_dump(by_alias=True))
```

Assuming the string "full_name" is only known / resolved at runtime, can I change the alias for the "name" field from "common_name" to "full_name" at runtime so that the output of model_dump shows:
{ "full_name" : "John", "age" : 200 }
?

I already have an existing method to do this but I would like to see if there is another way. The existing method users model_serializer in wrap mode.

**Comment by samuelcolvin** [comment, marked as answer]
Using `model_serialize` in wrap mode is the right solution, there's no other way.

**Reply by jesumyip** [reply]
Thank you.
